### PR TITLE
2D DRP Development Practices

### PR DESCRIPTION
**Pull request by hassanxp**

### CODE REVIEW

File: dev-practices/practices.rst [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+to fix your contribution. In that case, have a look at the Travis log for clues as to what the problem might
+be.
+
+As you push new commits to the pull request, Travis will automatically trigger new integration tests. Because
+Travis is triggered on GitHub pull requests, you should ensure you have pushed your work on a common ticket
+branch to all appropriate repos before making pull requests. If you want to signal Travis `not to
+automatically test a commit`_, add the text ``[ci skip]`` to your commit message.
+
+Unfortunately, due to Travis resource limitations, only 4 MB of logs can be generated. We therefore trap the
+build and test output and display only the last 100 lines when the process completes. If this makes it
```
Comment (PaulPrice) [inline dev-practices/practices.rst]: The code is now much less chatty, so this is no longer necessary: all output is displayed.

File: dev-practices/practices.rst [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+but has its own limitations in that it is project-specific. 
+In general, as labels are more flexible it is not recommended to use components.
+For short-term work, the developer may use the "epic" field, 
+as it does have the advantage that JIRA highlights issues associated to a given epic in scrum boards, 
+but it also has a significant limitation in that only one epic can be assigned to one ticket. 
+Because of that, this field will not be used for general project scheduling.
+
+In general, developers are encouraged to file issues. But please check existing open issues 
+in case the problem or activity has already been addressed,
+ and where appropriate, your manager, before filing.
```
Comment (PaulPrice) [inline dev-practices/practices.rst]: Leading space messes up the formatting.

File: dev-practices/practices.rst [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+When work has been completed, please use ``git rebase -i`` to reorganize commits into logical units. 
+Ensure that all unit tests and the integration test pass. 
+Then create a GitHub Pull Request and name one or more reviewers. 
+The pull request will trigger the continuous integration build. Check that the build has passed.
+Then finally set the issue status to “In Review” and assign the same reviewers to the issue as with the pull request.
+
+Please be aware that reviewers may be extremely busy so may not be able to review the issue immediately. 
+However, having a significant amount of code in branches pending to be merged for a long period of time is also problematic, 
+so to remedy such situations the developer should allow a maximum of 7 working days for the reviewer to take action.
```
Comment (PaulPrice) [inline dev-practices/practices.rst]: A week and a half is a significant fraction of the typical sprint.

**Review by PaulPrice — APPROVED** [review]
Looks reasonable to me.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,30 @@
+dev-practices/_build/doctrees/environment.pickle
```
Comment (PaulPrice) [inline .gitignore]: This file should go in the `dev-practises` directory, and probably needs to contain only the line:
```
_build
```
See the other doc directories for examples.

File: dev-practices/practices.rst (hunk 1 above)
Comment (hassanxp) [reply dev-practices/practices.rst]: Ok. Section deleted.

File: dev-practices/practices.rst (hunk 2 above)
Comment (hassanxp) [reply dev-practices/practices.rst]: Thanks. Fixed.

File: dev-practices/practices.rst (hunk 3 above)
Comment (hassanxp) [reply dev-practices/practices.rst]: Ok I've reduced this to 5 days which is what you initially proposed I believe. We will see if this is workable or not as we proceed.

File: .gitignore (hunk 4 above)
Comment (hassanxp) [reply .gitignore]: I had a look at your ``simulatorUser`` and ``pipelineUser`` directories and it seems there is the same setup (in that the pickle file is in the ``doctrees`` subfolder). Have I misunderstood something?

Comment (PaulPrice) [reply .gitignore]: I'm talking about the `.gitignore` file.

Comment (hassanxp) [reply .gitignore]: Sorry I missed that. Fixed.

**Pull request merged**